pool/pool.go

```diff
@@ -480,12 +551,40 @@ func (p *pool) ObjectPayloadRangeSHA256(ctx context.Context, params *client.Rang
 	if err != nil {
 		return nil, err
 	}
-	res, err := cp.client.ObjectPayloadRangeSHA256(ctx, params, options...)
+
+	// FIXME: pretty bad approach but we should not mutate params through the pointer
+	//  If non-SHA256 algo is set then we need to reset it.
+	params = copyRangeChecksumParams(params)
+	// SHA256 by default, no need to do smth
```

Comment:
I suggest not to copy params anywhere. If client functions provide side effects, then it's better to make them not to.

Reply:
In future `Client` methods won't accept pointer to the parameters but parameters itself. But for now there is no possibility to avoid copying.

Reply:
I missed that this function requires `SHA256` hash but `params` may define `TZ` and we need to update params structure using the pointer. Then it totally makes sense.